src/extractor/languages/javascript.ts

```diff
@@ -0,0 +1,155 @@
+const WHITESPACE = [' ', '\t', '\r', '\n', '\f', '\v'];
+const STRING_QUOTES = ["'", '"', '`'];
+const CONSUMABLE_MAP: Record<string, string> = {
+  "'": "'",
+  '"': '"',
+  '`': '`',
+  '{': '}',
+  '[': ']',
+  '(': ')',
+};
+
+type Result<T> = {
+  extracted: T;
+  consumed: number;
+};
+
+export function extractString(stringCode: string): string | null {
+  stringCode = stringCode.trim();
+  const concatRegex = new RegExp(`(?!\\\\).${stringCode[0]}\\s*\\+`);
+  if (
+    (stringCode.startsWith('`') && stringCode.match(/(?!\\).\${.*?}/)) ||
+    stringCode.match(concatRegex)
+  ) {
+    // Dynamic string
+    return null;
+  }
+
+  if (stringCode[0] !== stringCode[stringCode.length - 1]) {
+    return null;
+  }
+
+  return stringCode.slice(1, -1).replaceAll('\\', '');
+}
+
+function extractKeyName(key: string): string {
+  let decodedKey = key;
+  if (decodedKey.startsWith('[')) {
+    if (!STRING_QUOTES.includes(decodedKey[1])) return key;
```

Comment:
Please don't use `if` statements without parenthesis, it can be quite tricky to read.